src/main.cpp

```diff
@@ -1510,16 +1510,18 @@ bool AcceptToMemoryPool(
  * Return transaction in txOut, and if it was found inside a block, its hash is placed in hashBlock.
  * If blockIndex is provided, the transaction is fetched from the corresponding block.
  */
-bool GetTransaction(const uint256 &hash, CTransaction &txOut, const Consensus::Params& consensusParams, uint256 &hashBlock, bool fAllowSlow, CBlockIndex* blockIndex)
+bool GetTransaction(const uint256 &hash, CTransaction &txOut, const Consensus::Params& consensusParams, uint256 &hashBlock, bool fAllowSlow, uint32_t *pnBlockHeight, CBlockIndex* blockIndex)

```

Comment:
there are so many paths in that function with return statement.
You can use the following technique in this case:
bool bRet = false;
uint32_t nBlockHeight = numeric_limits<uint32_t>::max();
do
{
      code_block1 that sets(or doesn't) nBlockHeight, 
            break instead of return
      code_block2 that sets(or doesn't) nBlockHeight, 
            break instead of return
     ..................
} while (false);
if (pnBlockHeight)
   *pnBlockHeight = nBlockHeight;
return bRet;

that way you have full control over setting pnBlockHeight. Also it's much easier to debug the function which is written like this.